kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/FeedbackInputFragment.java

```diff
@@ -0,0 +1,196 @@
+    static final int RATINGBAD = 1;
+    static final int RATINGGOOD = 2;
+    static final int RATINGGREAT = 3;
+    private int ratingValue;
+
+    public void setFeedbackFragmentListener(FeedbackFragmentListener feedbackFragmentListener) {
+        this.feedbackFragmentListener = feedbackFragmentListener;
+    }
+
+    public FeedbackInputFragment() {
+        super();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (!FeedbackFragmentListener.class.isAssignableFrom(MobiComConversationFragment.class)) {
+            Utils.printLog(context, TAG, "Implement FeedbackFragmentListener in your parent fragment.");
+            throw new ClassCastException("Implement FeedbackFragmentListener in your parent fragment.");
+        }
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.feedback_rating_layout, container, false);
+
+        editTextFeedback = view.findViewById(R.id.idEditTextFeedback);
+        buttonSubmit = view.findViewById(R.id.idButtonSubmit);
+        button1 = view.findViewById(R.id.idButton1);
+        button2 = view.findViewById(R.id.idButton2);
+        button3 = view.findViewById(R.id.idButton3);
+        textView1 = view.findViewById(R.id.idText1);
+        textView2 = view.findViewById(R.id.idText2);
+        textView3 = view.findViewById(R.id.idText3);
+        closeButton = view.findViewById(R.id.idCloseFeedbackFragment);
+
+        button1.setScaleX(0.8f);
+        button2.setScaleX(0.8f);
+        button3.setScaleX(0.8f);
+        button1.setScaleY(0.8f);
+        button2.setScaleY(0.8f);
+        button3.setScaleY(0.8f);
+
+        editTextFeedback.setVisibility(View.GONE);
+        textView1.setVisibility(View.INVISIBLE);
+        textView2.setVisibility(View.INVISIBLE);
+        textView3.setVisibility(View.INVISIBLE);
+
+        buttonSubmit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(ratingValue == 0) {
+                    ratingValue = RATINGGOOD;
+                }
+                String feedbackComment = editTextFeedback.getText().toString().trim();
+                feedbackFragmentListener.onFeedbackSubmitButtonPressed(ratingValue, feedbackComment);
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        closeButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        button1.setOnClickListener(new View.OnClickListener() {
+            @Override
```

Comment:
Looks like duplicate code for each button, figure out a better way

Reply:
used a single listener